sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java

```diff
@@ -59,31 +65,120 @@ public HttpPipeline getHttpPipeline() {
         return this.protocolLayer.getHttpPipeline();
     }
 
-    // TODO These are temporary implementations for sample purposes. This should be spruced up/replaced once this API is actually designed.
-    /**
-     * Creates a digital twin.
-     *
-     * @param digitalTwinId The Id of the digital twin.
-     * @param digitalTwin The application/json digital twin to create.
-     * @return The application/json digital twin created.
-     */
+    // Input and output as String.
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<String> createDigitalTwin(String digitalTwinId, String digitalTwin) {
-        try {
-            return protocolLayer
-                .getDigitalTwins()
-                .addWithResponseAsync(digitalTwinId, digitalTwin)
-                // Mono.flatMap: Transform the item emitted by this Mono asynchronously, returning the value emitted by another Mono (possibly changing the value type).
-                // The PL gives us a Mono<DigitalTwinsAddResponse>, so we use Mono.flatMap to transform the items emitted
-                // from Mono<DigitalTwinsAddResponse> to Mono<String>, asynchronously.
-                .flatMap(
-                    // Mono.just(item) creates a new Mono that emits the specified item.
-                    // response.getValue gives us the deserialized Http response body (Object).
-                    response -> Mono.just(response.getValue().toString()));
-        } catch (RuntimeException ex) {
-            // TODO: Ensure that exceptions are handled in a reactive way
-            return FluxUtil.monoError(logger, ex);
-        }
+    public Mono<String> createDigitalTwinString(String digitalTwinId, String digitalTwin) throws JsonProcessingException {
+        Object payload = mapper.readValue(digitalTwin, Object.class);
+        return protocolLayer
+            .getDigitalTwins()
+            .addWithResponseAsync(digitalTwinId, payload)
+            .flatMap(
+                response -> {
+                    try {
+                        return Mono.just(mapper.writeValueAsString(response.getValue()));
+                    } catch (JsonProcessingException e) {
+                        return Mono.error(e);
+                    }
+                });
+    }
+
+    // Input and output are Object.
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Object> createDigitalTwinObject(String digitalTwinId, Object digitalTwin) {
+        return protocolLayer
+            .getDigitalTwins()
+            .addWithResponseAsync(digitalTwinId, digitalTwin)
+            .flatMap(
+                response -> Mono.just(response.getValue()));
+    }
+
+    // Input and output are T (Generics).
+    @ServiceMethod(returns = ReturnType.SINGLE)
```

Comment:
Since you have done more research into these, what are the pros and/or cons of this vs other 2 approaches?